core/src/main/java/org/apache/iceberg/PartitionStats.java

```diff
@@ -249,4 +250,45 @@ public <T> void set(int pos, T value) {
         throw new UnsupportedOperationException("Unknown position: " + pos);
     }
   }
+
+  @Override
+  @SuppressWarnings("checkstyle:CyclomaticComplexity")
+  public boolean equals(Object other) {
```

Comment:
`StructLikeMap` was previously handling this implicitly. But when `PartitionStatsRecord` wraps `PartitionStats` now for the writers, it needs to override `equals` and `hashcode`